common/key_value_parser/key_value_parser.cpp

```diff
@@ -49,6 +51,15 @@ error::Error MakeError(KeyValueParserErrorCode code, const string &msg) {
 
 ExpectedKeyValuesMap ParseKeyValues(const vector<string> &items, char delimiter) {
 	KeyValuesMap ret;
+	error::Error err = AddParseKeyValues(ret, items, delimiter);
+	if (err) {
+		return ExpectedKeyValuesMap(err);
+	} else {
+		return ExpectedKeyValuesMap(ret);
```

Comment:
I think you can just write `return err` and `return ret` here.

Reply:
How does that work? I saw `return true;` in a function returning `ExpectedBool()`, but no idea how/why it works.

Reply:
Since the compiler already knows that the return type of the function is `ExpectedKeyValuesMap`, it just puts two and two together and transforms `return <RET>` into `return ExpectedKeyValuesMap(<RET>)` for you. I suppose one could argue that it's less explicit, but I think that's fine as long as we name our variables sensibly.

Reply:
Like I said above in https://github.com/mendersoftware/mender/pull/1152#discussion_r1115568345, I don't like this so I'd prefer not to use it.